teamcity: failed tests on master: testrace/TestStoreRecoverWithErrors

The following tests appear to have failed:

[#216908](https://teamcity.cockroachdb.com/viewLog.html?buildId=216908):

```
--- FAIL: testrace/TestStoreRecoverWithErrors (0.080s)
client_raft_test.go:211: key range "b"-"b\x00" outside of bounds of range /Min-/System/""
------- Stdout: -------
I170411 19:06:40.323179 12261 gossip/gossip.go:293  [n1] NodeDescriptor set to node_id:1 address:<network_field:"" address_field:"" > attrs:<> locality:<> 
I170411 19:06:40.330387 12261 storage/store.go:1256  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170411 19:06:40.339009 12356 storage/split_queue.go:92  [split,s1,r1/1:/M{in-ax},@c4204f0000] splitting at key /System/""
I170411 19:06:40.352204 12356 storage/replica_command.go:2602  [split,s1,r1/1:/M{in-ax},@c4204f0000] initiating a split of this range at key /System/"" [r2]
I170411 19:06:40.361011 12261 util/stop/stopper.go:494  quiescing; tasks left:
1      storage/queue.go:523
1      kv/txn_coord_sender.go:903
I170411 19:06:40.363205 12261 util/stop/stopper.go:494  quiescing; tasks left:
1      storage/queue.go:523
W170411 19:06:40.368058 12344 storage/store.go:1458  [split,s1,r1/1:/M{in-ax},@c4204f0000] unable to gossip on capacity change: node unavailable; try another peer
I170411 19:06:40.371117 12261 gossip/gossip.go:293  [n1] NodeDescriptor set to node_id:1 address:<network_field:"" address_field:"" > attrs:<> locality:<> 
I170411 19:06:40.376783 12261 storage/store.go:1256  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170411 19:06:40.382217 12160 storage/split_queue.go:92  [split,s1,r2/1:/{System/""-Max},@c4204f0c00] splitting at key /System/tsd
I170411 19:06:40.382872 12261 util/stop/stopper.go:494  quiescing; tasks left:
2      storage/replica.go:4429
2      gossip/infostore.go:301
1      storage/replica_range_lease.go:237
1      storage/queue.go:523
I170411 19:06:40.383085 12261 util/stop/stopper.go:494  quiescing; tasks left:
2      storage/replica.go:4429
2      gossip/infostore.go:301
1      storage/queue.go:523
W170411 19:06:40.383195 12206 storage/id_alloc.go:109  [s1] node unavailable; try another peer
E170411 19:06:40.383471 12160 storage/queue.go:634  [split,s1,r2/1:/{System/""-Max},@c4204f0c00] unable to split [n1,s1,r2/1:/{System/""-Max}] at key "/System/tsd": storage/replica_command.go:2594: unable to allocate right hand side range descriptor: could not allocate ID; system is draining
I170411 19:06:40.383602 12261 util/stop/stopper.go:494  quiescing; tasks left:
2      storage/replica.go:4429
2      gossip/infostore.go:301
I170411 19:06:40.383778 12261 util/stop/stopper.go:494  quiescing; tasks left:
2      storage/replica.go:4429
1      gossip/infostore.go:301
I170411 19:06:40.384072 12261 util/stop/stopper.go:494  quiescing; tasks left:
1      storage/replica.go:4429

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -171,10 +171,11 @@ func TestStoreRecoverWithErrors(t *testing.T) {
 		stopper := stop.NewStopper()
 		defer stopper.Stop()
 		storeCfg := storeCfg // copy
+		keyA := rg1Key("a")
 		storeCfg.TestingKnobs.TestingEvalFilter =
 			func(filterArgs storagebase.FilterArgs) *roachpb.Error {
 				_, ok := filterArgs.Req.(*roachpb.IncrementRequest)
-				if ok && filterArgs.Req.Header().Key.Equal(roachpb.Key("a")) {
+				if ok && filterArgs.Req.Header().Key.Equal(keyA) {
 					numIncrements++
 				}
 				return nil
@@ -182,14 +183,14 @@ func TestStoreRecoverWithErrors(t *testing.T) {
 		store := createTestStoreWithEngine(t, eng, true, storeCfg, stopper)
 
 		// Write a bytes value so the increment will fail.
-		putArgs := putArgs(roachpb.Key("a"), []byte("asdf"))
+		putArgs := putArgs(keyA, []byte("asdf"))
 		if _, err := client.SendWrapped(context.Background(), rg1(store), putArgs); err != nil {
 			t.Fatal(err)
 		}
 
 		// Try and fail to increment the key. It is important for this test that the
 		// failure be the last thing in the raft log when the store is stopped.
-		incArgs := incrementArgs(roachpb.Key("a"), 42)
+		incArgs := incrementArgs(keyA, 42)
 		if _, err := client.SendWrapped(context.Background(), rg1(store), incArgs); err == nil {
 			t.Fatal("did not get expected error")
 		}
@@ -206,7 +207,8 @@ func TestStoreRecoverWithErrors(t *testing.T) {
 	store := createTestStoreWithEngine(t, eng, false, storeCfg, stopper)
 
 	// Issue a no-op write to lazily initialize raft on the range.
-	incArgs := incrementArgs(roachpb.Key("b"), 0)
+	keyB := rg1Key("b")
+	incArgs := incrementArgs(keyB, 0)
 	if _, err := client.SendWrapped(context.Background(), rg1(store), incArgs); err != nil {
 		t.Fatal(err)
 	}

--- a/pkg/storage/client_replica_test.go
+++ b/pkg/storage/client_replica_test.go
@@ -183,7 +183,7 @@ func TestTxnPutOutOfOrder(t *testing.T) {
 	// key is selected to fall within the meta range in order for the later
 	// routing of requests to range 1 to work properly. Removing the routing
 	// of all requests to range 1 would allow us to make the key more normal.
-	const key = "\x03key"
+	key := rg1Key("key")
 	// Set up a filter to so that the get operation at Step 3 will return an error.
 	var numGets int32
 

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -25,6 +25,7 @@ client_*.go.
 package storage_test
 
 import (
+	"bytes"
 	"fmt"
 	"math/rand"
 	"net"
@@ -77,6 +78,15 @@ func rg1(s *storage.Store) client.Sender {
 		}
 		return ba
 	})
+}
+
+// rg1Key returns a key that will reliably be located within the first range.
+// This is a bit of a bandaid to ensure tests continue working as intended
+// until the rg1 method is removed.
+func rg1Key(key string) roachpb.Key {
+	// Ensure that the key will be located in the first range by giving it the
+	// meta2 prefix. The first range only contains meta keys as of April 2017.
+	return roachpb.Key(bytes.Join([][]byte{keys.Meta2Prefix, []byte(key)}, nil))
 }
 
 // createTestStore creates a test store using an in-memory

